src/Polysemy/Internal/Strategy.hs

```diff
@@ -2,130 +2,78 @@
 
 module Polysemy.Internal.Strategy where
 
+import Polysemy.Final.Type
 import Polysemy.Internal
-import Polysemy.Internal.Combinators
-import Polysemy.Internal.Tactics (Inspector(..))
-
-
-
-data Strategy m f n z a where
-  GetInitialState     :: Strategy m f n z (f ())
-  HoistInterpretation :: (a -> n b) -> Strategy m f n z (f a -> m (f b))
-  GetInspector        :: Strategy m f n z (Inspector f)
+import Polysemy.Internal.Union
+import Polysemy.Internal.Tactics
 
 
 ------------------------------------------------------------------------------
 -- | 'Strategic' is an environment in which you're capable of explicitly
 -- threading higher-order effect states to the final monad.
--- This is a variant of @Tactics@ (see 'Polysemy.Tactical'), and usage
--- is extremely similar.
+-- 'Strategic' makes use of @Tactics@ (see 'Polysemy.Tactical'),
+-- and you can use the same methods as @Tactics@ (such as 'Polysemy.runT'
+-- and 'Polysemy.bindT') in order to convert actions of @m@ to the final monad.
 --
 -- @since 1.2.0.0
-type Strategic m n a = forall f. Functor f => Sem (WithStrategy m f n) (m (f a))
+type Strategic m n a = forall f. Functor f => Sem (WithStrategy m f n) (f a)
 
 
 ------------------------------------------------------------------------------
 -- | @since 1.2.0.0
-type WithStrategy m f n = '[Strategy m f n]
+type WithStrategy m f n = '[Tactics m f n, Embed m, Final m]
 
 
 ------------------------------------------------------------------------------
 -- | Internal function to process Strategies in terms of
 -- 'Polysemy.Final.withWeavingToFinal'.
 --
 -- @since 1.2.0.0
-runStrategy :: Functor f
-            => Sem '[Strategy m f n] a
+runStrategy :: forall m f n a
+             . (Functor f, Monad m)
+            => Sem '[Tactics m f n, Embed m, Final m] a
```

Comment:
@KingOfTheHomeless, the suggestion is to replace `runM` with this thing, right?

Reply:
No, `runM` shouldn't have anything to do with tactics: the suggestion was to have `runM` like this:

```haskell
runM :: Monad m => Sem '[Embed m, Final m] a -> m a
runM = usingSem $ \u -> case decomp u of
  Right (Weaving (Embed m) s _ ex _) ->
    (ex . (<$ s)) <$> m
  Left g -> case extract g of
    Weaving (WithWeavingToFinal wav) s wv ex ins ->
      ex <$> wav s (runM . wv) ins
```

If we do, then `runStrategy` can be implemented like this:
```haskell
runStrategy sem = \s wv ins -> runM $ interpret
  (\case
    GetInitialState       -> pure s
    HoistInterpretation f -> pure $ \fa -> wv (f <$> fa)
    GetInspector          -> pure (Inspector ins)
  ) sem
```